packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -151,7 +179,11 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
       final String? fileName = imageMap['filename'] as String?;
 
       if (size != null && idiom != null && scale != null && fileName != null) {
-        iconInfo['$idiom $size $scale'] = fileName;
+        // for example, "64x64". Parse the width since it is a square.
+        final double parsedSize = size.split('x').map((String element) => double.parse(element)).first;
+        // for example, "3x". Remove 'x' and parse int.
+        final int parsedScale = int.parse(scale.substring(0, scale.length-1));
```

Comment:
can you trim the x off it's there to guard against some case where they stop adding an x?  Can scale be an empty string?

Reply:
From [the spec](https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_ref-Asset_Catalog_Format/ImageSetType.html#//apple_ref/doc/uid/TP40015170-CH25-SW3) it always has `x` suffix (or the whole tag is ignored if it's using pdf rather than png). It will not have empty string.